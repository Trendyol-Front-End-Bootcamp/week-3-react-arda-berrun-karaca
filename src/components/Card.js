import React from 'react';
import { Link } from 'react-router-dom';

import './styles/Card.css';

export const addStatusClass = (status) => {
  const newStatus = status?.toLowerCase();
  if (newStatus === 'alive' || newStatus === 'dead') {
    return `status-${newStatus}`;
  }

  return 'status-unknown';
};

function Card({ id, name, status, species, gender, image, location }) {
  return (
    <article className="card">
      <div className="card-image">
        <img src={image} alt={name} />
      </div>
      <div className="card-content">
        <div className="card-content-section">
          <Link to={`/character/${id}`}>
            <h3>{name}</h3>
          </Link>
          <span className="info">
            <span
              name="status-info"
              className={`status ${addStatusClass(status)}`}
            ></span>
            {status} - {species} - {gender}
          </span>
        </div>
        <div className="card-content-section">
          <span>Last known location:</span>
          <p>{location}</p>
        </div>
      </div>
    </article>
  );
}

export default Card;
