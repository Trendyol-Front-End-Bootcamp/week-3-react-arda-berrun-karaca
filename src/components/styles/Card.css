.card {
  background-color: var(--card-background);
  color: var(--main-white);
  display: flex;
  flex-direction: column;
  border-radius: 10px;
  overflow: hidden;
}

.card-image {
  flex: 2 1 0%;
  /* width: 100%; */
}

.card-image img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.card-content {
  flex: 3 1 0%;
  /* width: 100%; */
}

.card-content-section a {
  color: var(--main-white);
  text-decoration: none;
}
.card-content-section:last-child span {
  color: var(--secondary-gray);
}
.card-content-section:last-child {
  margin-top: 2rem;
  line-height: 1.6;
}
@media screen and (min-width: 576px) {
  .card {
    width: 500px;
    height: 220px;
    flex-direction: row;
  }
}

.info {
  position: relative;
  margin-left: 1rem;
}
.status {
  display: block;
  width: 10px;
  height: 10px;
  border-radius: 50%;
  position: absolute;
  top: 3px;
}

.status-alive {
  background-color: var(--status-alive);
}

.status-unknown {
  background-color: var(--status-unknown);
}
.status-dead {
  background-color: var(--status-dead);
}

.card-content {
  padding: 0.5rem 1rem;
  line-height: 1.6;
}

.card-content h3 {
  font-size: 1.5rem;
  cursor: pointer;
}
.card-content h3:hover {
  color: var(--dark-green);
}
