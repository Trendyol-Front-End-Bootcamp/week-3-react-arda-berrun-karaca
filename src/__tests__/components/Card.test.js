import React from 'react';
import renderer from 'react-test-renderer';

import Card, { addStatusClass } from '../../components/Card';
import {BrowserRouter as Router} from 'react-router-dom';

describe('Card tests', () => {
  it('should be defined', () => {
    expect(Card).toBeDefined();
  });

  it('should render correctly', () => {
    const props = {
        id: 1,
        name: 'Rick Sanchez',
        status: 'Alive',
        species: 'Human',
        gender: 'Male',
        image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
        location: 'Earth'
    };

    const wrapper = renderer.create(
      <Router>
        <Card {...props} />
      </Router>
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should be an element with the status-alive class', () => {
    expect(addStatusClass('Alive')).toEqual('status-alive');
  });

  it('should be an element with the status-dead class', () => {
    expect(addStatusClass('Dead')).toEqual('status-dead');
  });

  it('should be an element with the status-unknown class', () => {
    expect(addStatusClass('dlkdak')).toEqual('status-unknown');
    expect(addStatusClass()).toEqual('status-unknown');
  });
});
