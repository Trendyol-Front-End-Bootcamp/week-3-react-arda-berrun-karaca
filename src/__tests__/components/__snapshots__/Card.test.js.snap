// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests should render correctly 1`] = `
<article
  className="card"
>
  <div
    className="card-image"
  >
    <img
      alt="Rick Sanchez"
      src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    />
  </div>
  <div
    className="card-content"
  >
    <div
      className="card-content-section"
    >
      <a
        href="/character/1"
        onClick={[Function]}
      >
        <h3>
          Rick Sanchez
        </h3>
      </a>
      <span
        className="info"
      >
        <span
          className="status status-alive"
          name="status-info"
        />
        Alive
         - 
        Human
         - 
        Male
      </span>
    </div>
    <div
      className="card-content-section"
    >
      <span>
        Last known location:
      </span>
      <p>
        Earth
      </p>
    </div>
  </div>
</article>
`;
